notes

```diff
@@ -173,10 +173,17 @@ handle_multiple_notes() {
 get_full_note_path() {
     local note_path=$1
 
-    if [[ "$note_path" != *.$NOTES_EXT ]]; then
-        note_path="$note_path.$NOTES_EXT"
-    fi
-    if [ ! -f "$note_path" ]; then
+    # first check if file exists
+    if [ -f "$note_path" ]; then # note path given is good
+        note_path="$note_path"
+    elif [ -f "$notes_dir/$note_path" ]; then # exists
+        note_path="$notes_dir/$note_path"
+    elif echo "$note_path" | grep '[.][A-Za-z]\{1,4\}$' &>/dev/null; then # has a 1-4 letter extension
+        note_path="$notes_dir/$note_path"
+    else
+        if [[ "$note_path" != *.$NOTES_EXT ]]; then
+            note_path="$note_path.$NOTES_EXT"
+        fi
         note_path="$notes_dir/$note_path"
```

Comment:
Makes sense to me. I don't use periods in any of my filenames other than before the ext, so I didn't really think of this and just went with what seemed to work and not break anything else.

Reply:
I made some changes and I think it should look for matches in the order you suggested now.

It still passed all the tests run by test.sh, but playing around a little, it seems that the `remove_note` function would have to be changed too. It doesn't call `get_full_note_path`.  If you have `note.test` and `note.test.md` and you do `notes rm note.test`, it will delete `note.test.md`

Probably a pretty unlikely scenario though.

Probably, remove_note just needs to check if the argument is a directory and if not, use `get_full_note_path`

remove_note doesn't look like it accounts for absolute or relative paths... 

There may be a few other functions that don't use `get_full_note_path`. I might search through the code and look for other potential problems later on.

Reply:
hacked remove_note. Didn't find any other functions that skip `get_full_note_path`.